salt/modules/yumpkg.py

```diff
@@ -1769,6 +1777,8 @@ def _temporarily_unhold(pkgs, targets):
                 cmd.extend(["--best", "--allowerasing"])
             _add_common_args(cmd)
             cmd.append("install" if pkg_type != "advisory" else "update")
+            if _yum() == "dnf5":
```

Comment:
Why aren't we just including this logic above at line 1776: `if _yum() in ("dnf", "dnf5")` since its adding the same args?

Reply:
Because dnf5 objects to having --allowerasing as a general option now, in this case it is only allowed as an option after the ```install`` command.

dnf5 has changed options around, and this --allowerasing after the command also applies to other commands too, just we don't use them in yumpkg.py. See the original issue which explains how dnf5 has changed things.  Went through the full dnf5 changes to make sure there were not other issues missed, for example: dnf5 does not have a 'needs_restart' option.

Reply:
Thanks for confirming.